lisa/util/parallel.py

```diff
@@ -51,6 +104,7 @@ def wait_worker(self) -> bool:
             self._futures.remove(future)
             # exception will throw at this point
             self._callback(result)
```

Comment:
The closed one may need to be removed from the map. It may create a reference, so some resource are not recycled.